src/main/java/org/spongepowered/api/CatalogType.java

```diff
@@ -35,7 +37,7 @@
  * <code>`a.getId().equalsIgnoreCase(b.getId())`</code> are true then all must
  * be true.</p>
  */
-public interface CatalogType {
+public interface CatalogType extends TextRepresentable {
```

Comment:
It does not **need** to, but it can due to its nature. It has a human readable name and and thus should be displayed using said name (or the translation). Instead of `toString()`. This also allows plugins to use it on `CatalogType`s directly without having to check for it..(That's why I added the default impl in the first place).

Having this as CatalogType simply assures, that you can use it with commands/autocompletion.
`getName()` assures that the player can read it and `getId()` ensures that you can obtain it from the registry after command completion. Furthermore it does not cost us anything or prevents us from doing something else (especially if using a default impl).

I also thought about adding `TextRepresentable` to `Translation` and `Translatable` to further simplify `Text.of(Object...)` (Because in both cases everything those methods do is wrapping the `Translation` in a `TranslatableText`.)